feathr_project/feathr/client.py

```diff
@@ -251,7 +256,27 @@ def _construct_redis_client(self):
         self.logger.info('Redis connection is successful and completed.')
         self.redis_clint = redis_clint
 
-    def join_offline_features(self, feature_join_conf_path='feature_join_conf/feature_join.conf'):
+
+    def get_offline_features(self,
+                             observation_settings: ObservationSettings,
+                             feature_query: Union[FeatureQuery, List[FeatureQuery]]):
+        # produce join config
+        tm = Template("""
+            {{observation_settings.to_config()}}
+            featureList: [
+                {% for list in feature_lists %}
+                    {{list.to_config()}}
+                {% endfor %}
+            ]
+        """)
+        feature_queries = feature_query if isinstance(feature_query, List) else [feature_query]
+        config = tm.render(feature_lists=feature_queries, observation_settings=observation_settings)
```

Comment:
I assume even if I don't register the features, the offline features will save all the feature objects into templates and submit the spark job?

Reply:
Updated